generator/src/main/java/com/scottlogic/deg/generator/generation/StandardFieldValueSourceEvaluator.java

```diff
@@ -62,15 +65,12 @@ private boolean mustBeNull(FieldSpec fieldSpec) {
     }
 
     private List<FieldValueSource> getSetRestrictionSources(FieldSpec fieldSpec) {
-        Stream<Object> whitelist = fieldSpec.getSetRestrictions().getWhitelist().stream();
-
-        if (mayBeNull(fieldSpec)) {
-            return Arrays.asList(
-                new CannedValuesFieldValueSource(whitelist.collect(Collectors.toList())),
-                nullOnlySource);
-        }
+        List<Object> whitelist = new ArrayList<>(fieldSpec.getSetRestrictions().getWhitelist());
+         if (whitelist.isEmpty()){
+             return new LinkedList<>();
+         }
 
-        return Collections.singletonList(new CannedValuesFieldValueSource(whitelist.collect(Collectors.toList())));
+        return new LinkedList<>(Arrays.asList(new CannedValuesFieldValueSource(whitelist)));
```

Comment:
Why not use `Collections.singletonList`?

Reply:
this is why i asked for the re review :P

this list needs to be mutable to allow adding the null source.

Though i could probably use new LinkedList<>(Collections.singletonList(...))

Not sure what's best to do here

Reply:
How about this:

```
return Collections.singletonList(...)
```

and add a function to append/concat to a list as a new list

```
private static T List<T> concat(List<T> collection, T... append){
   ArrayList<T> newList = new ArrayList<>(collection);
   newList.addAll(append);
   return newList;
}
```

then use
```
List<FieldValueSource> setRestrictionSources = getSetRestrictionSources(fieldSpec);
return mayBeNull(fieldSpec)
   ? concat(setRestrictionSources, nullOnlySource)
   : setRestrictionSources
```